libtenzir/builtins/operators/python.cpp

```diff
@@ -128,10 +130,15 @@ class python_operator final : public crtp_operator<python_operator> {
         // TODO: Handle broken venvs. Maybe there is a way to check whether the
         // list of requirements is installed correctly?
         if (!exists(venv_path, ec)) {
-          if (bp::system(python_executable, "-m", "venv", venv)) {
-            ctrl.abort(caf::make_error(ec::system_error,
-                                       "failed to create virtualenv "
-                                       "(see log for details)"));
+          // The default size of the pipe buffer is 64k on Linux, we assume
+          // (hope) that this is enough. A possible solution would be to wrap
+          // the invocation in a script that drains the pipe continuously but
+          // only forwards the first n bytes.
```

Comment:
What happens if the buffer is too small?

Reply:
The subprocess blocks in a call to `write`.

Reply:
What does it mean for the operator? Does it stall entirely, fail, or is it just blocking the thread for a while?

Reply:
I think it's very unlikely that either process writes that much to stderr.

Reply:
Right, but what would that mean for the operator? In case a process does this, do we hang, crash, or fail? I think that should be explained in the comment.

Reply:
I wrote an issue for this: https://github.com/tenzir/issues/issues/1079